File: Week2-Arrays/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>


int countLetters(string s);
int countWords(string s);
int countSentences(string s);

int main(void) {
  string s = get_string("Text: ");
  int letters, words, sentences;
  float index;
  
  letters = countLetters(s);
  words = countWords(s);
  sentences = countSentences(s);

  float letterPerWords = (float) letters / words * 100;
  float sentencesPerWords = (float) sentences / words * 100;
  
  index = 0.0588 * letterPerWords - 0.296 * sentencesPerWords - 15.8;
  int grade = round(index);

  if (grade < 1) {
    printf("Before Grade 1\n");
  } else if (grade >= 16) {
    printf("Grade 16+\n");
  } else {
    printf("Grade %d\n", grade);
  }
  
  return 0;
}

int countLetters(string s) {
  int count = 0;
  for (int i = 0; s[i] != '\0'; i++) {
    if (isalpha(s[i])) {
      count++;
    }
  }

  return count;
}

int countWords(string s) {
  int words = 0;
  bool inWord = false;

  for (int i = 0; s[i] != '\0'; i++) {
    if (isalpha(s[i]) || s[i] == '\'') {
      if (!inWord) {
        inWord = true;
        words++;
      } else {
        continue;
      }
    } else {
      inWord = false;
    }
  }
  return words;
}

int countSentences(string s) {
  int sentences = 0;
  
  for (int i = 0; s[i] != '\0'; i++) {
    if (s[i] == '.' || s[i] == '!' || s[i] == '?') {
      sentences++;
    }
  }

  return sentences;
}

File: week-6/readability.py
def count_letters(text):
    letters = 0
    for c in text:
        if c.isalpha():
            letters += 1
    return letters

def count_words(text):
    return len(text.split())

def count_sentences(text):
    sentences = 0
    for c in text:
        if c in ".!?":
            sentences += 1
    return sentences

while True:
    text = input("Text: ")
    if len(text) > 0:
        break
    

letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

letter_per_words = letters / words * 100
sentences_per_words = sentences / words * 100

grade = round(0.0588 * letter_per_words - 0.296 * sentences_per_words - 15.8)
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
